Cargo.lock

```diff
@@ -7236,8 +7243,26 @@ dependencies = [
  "num-derive",
  "num-traits",
  "num_enum",
- "solana-program 1.14.8",
- "solana-zk-token-sdk 1.14.8",
+ "solana-program 1.15.1",
+ "solana-zk-token-sdk 1.15.1",
+ "spl-memo",
+ "spl-token",
+ "thiserror",
+]
+
+[[package]]
+name = "spl-token-2022"
+version = "0.6.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "67fcd758e8d22c5fce17315015f5ff319604d1a6e57a73c72795639dba898890"
+dependencies = [
+ "arrayref",
+ "bytemuck",
+ "num-derive",
+ "num-traits",
+ "num_enum",
+ "solana-program 1.15.1",
+ "solana-zk-token-sdk 1.15.1",
```

Comment:
I think this explains the recompilations, since we have spl-token-2022 0.5.0 and 0.6.0 with this change. If you want, push a new version of spl-associated-token-account that builds against spl-token-2022 0.6.0, and update the monorepo to also use that. That should reduce the recompilations